src/include/session.h

```diff
@@ -138,6 +138,21 @@ struct __wt_session_impl {
     } * scratch_track;
 #endif
 
+    /*
+     * Record the important timestamps of each stage in an eviction. If an eviction takes a long
+     * time and times out, we can trace the time usage of each stage from this information.
+     */
+    struct __wt_evict_timeline {
+        uint64_t evict_start;
+#ifdef HAVE_DIAGNOSTIC
```

Comment:
Why these time stats are gathered only in debug build? Collecting the time can have an overhead of around 60 to 100 microseconds per call based on the internal API. Also we are gathering this information for an eviction thread, so I feel it is fine to have that in the production build also.

Reply:
Application thread will also do eviction. Yes, may be that should also be fine. I was thinking in production we only gather the total time and in debug we gather more detailed break downs.

Reply:
Given the relatively long time to do eviction, gathering all these stats should be fine. I'll gather them in production as well.